antsibull/galaxy.py

```diff
@@ -183,6 +183,17 @@ async def download(self, collection: str, version: t.Union[str, semver.Version],
         :arg version: Version of the collection to download.
         :returns: The full path to the downloaded collection.
         """
+        # First check cache to avoid any network access
+        if self.collection_cache:
+            cache_filename = f"${collection.replace('.', '-')}-${version}.tar.gz"
+            if cache_filename in os.listdir(self.collection_cache):
+                # TODO: PY3.8: We can use t.Final in __init__ instead of cast here.
+                cached_copy = os.path.join(t.cast(str, self.collection_cache),
+                                           cache_filename)
+                download_filename = os.path.join(self.download_dir, cache_filename)
+                shutil.copyfile(cached_copy, download_filename)
+                return download_filename
```

Comment:
I don't think I want the cache to operate like this... I think I do want to have a network operation to retrieve info about the collection.  Then check the cache for a tarball matching the collection info.

Reply:
This is mainly useful for the ACD changelog generation, since with this I can run the changelog generation without any network access at all (once the cache is populated). But yeah, this caching here is more like a hack (since it is guessing the filename and not verifying the hash); I'll remove it.